generate_emoji_html.py

```diff
@@ -343,17 +343,42 @@ def _get_keys(dir_infos, limit, all_emoji, emoji_sort):
 
 
 def _parse_annotation_file(afile):
-  annotations = set()
-  line_re = re.compile(r'([0-9a-f ]+)')
+  """Parse file and return a map from sequences to one of 'ok', 'warning',
+  or 'error'.
+
+  The file format consists of two kinds of lines.  One defines the annotation
+  to apply, it consists of the text 'annotation:' followed by one of 'ok',
+  'warning', or 'error'.  The other defines a sequence that should get the most
+  recently defined annotation, this is a series of codepoints expressed in hex
+  separated by spaces.  The initial default annotation is 'error'.  '#' starts
+  a comment to end of line, blank lines are ignored.
+  """
+
+  annotations = {}
+  line_re = re.compile(r'annotation:\s*(ok|warning|error)|([0-9a-f ]+)')
```

Comment:
'((?:[0-9a-f]{4,6})(?: [0-9a-f]{4,6})*)' could be used, but I guess you thought it's an overkill.